easypost/__init__.py

```diff
@@ -902,6 +903,63 @@ def api_keys(self):
 
         return my_api_keys
 
+class Report(AllResource, CreateResource):
+    report_types = { 'shprep': 'shipment', 'plrep': 'payment_log', 'trkrep': 'tracker' }
+
+    @classmethod
+    def create(cls, api_key=None, **params):
+        requestor = Requestor(api_key)
+        url = cls.class_url()
+        wrapped_params = {cls.class_name(): params}
+        report_types = ['shipment', 'payment_log', 'tracker']
+
+        if str(params['type']) in report_types.values():
+            url += "/%s" % params['type']
+        else:
+            raise Exception("Undertermined Report Type")
+
+        response, api_key = requestor.request('post', url, wrapped_params, False)
+        return convert_to_easypost_object(response, api_key)
+
+    @classmethod
+    def retrieve(cls, easypost_id="", api_key=None, **params):
+        try:
+            easypost_id = easypost_id['id']
+        except (KeyError, TypeError):
+            pass
+
+        url = cls.class_url()
+
+        obj_id = params['public_id'].split("_")[0]
+
+        if obj_id in report_types  :
+            url += "/%s/%s" % (report_types[obj_id], params['public_id'])
+        else:
+            raise Exception("Undetermined Report Type")
+
+        if easypost_id == "":
+            requestor = Requestor(api_key)
+            response, api_key = requestor.request('get', url)
+            return convert_to_easypost_object(response, api_key)
+        else:
+            instance = cls(easypost_id, api_key, **params)
+            instance.refresh()
+            return instance
+
+    @classmethod
+    def all(cls, api_key=None, **params):
+        requestor = Requestor(api_key)
+        url = cls.class_url()
+        report_types = ['shipment', 'payment_log', 'tracker']
+
+        if str(params['type']) in report_types:
```

Comment:
Hmm this one doesn't call `.values()`, but I still think you can remove the redefined `report_types` array.